src/wasi_context.rs

```diff
@@ -114,6 +122,11 @@ impl WasiCtxBuilder {
             .iter()
             .map(|s| CString::new(s.as_bytes()).unwrap())
             .collect::<Vec<CString>>();
+        self.args_cstr_ptrs = self
+            .args
+            .iter()
+            .map(|s| s.as_ptr() as *const c_char)
+            .collect::<Vec<*const c_char>>();
```

Comment:
Should also apply similar change for self.env_cstr_ptrs in `pub fn set_env_vars(mut self, envs: Vec<&str>)`